client/js/jsonedit.js

```diff
@@ -1237,38 +1236,112 @@ function jeColorize() {
   $('#jeTextHighlight').innerHTML = out.join('\n');
 }
 
+/* Displaying and controlling tree subpane of edit area */
+
+const editPanel = document.getElementById("jeEditArea");
+
+let mouse_pos;
+
+function resize(e){
+  function pad(w) {
+    return parseInt(window.getComputedStyle(document.getElementById("jeEditArea")).getPropertyValue(w).slice(0,-2))
+  };
+  const panelHeight = editPanel.clientHeight;
+  const resizeHeight = document.getElementById("jeResize").clientHeight;
+  const headerHeight = document.getElementById("jeEditHeader").clientHeight;
+  const treeHeight = document.querySelector('#jeTree').clientHeight;
+  const padding = pad('padding-top') + pad('padding-bottom');
+
+  const height = Math.max(0,Math.min(panelHeight - resizeHeight - headerHeight - padding, treeHeight + e.y - mouse_pos));
+  editPanel.style.setProperty('--treeHeight', height + "px"); 
+  mouse_pos = e.y;
+}
+
+editPanel.addEventListener("mousedown", function(e){
+  if(e.target == $('#jeResize')) {
+    mouse_pos = e.y;
+    document.addEventListener("mousemove", resize, false);
+  }
+});
+
+document.addEventListener("mouseup", function(){
+  document.removeEventListener("mousemove", resize, false);
+});
+
 function jeDisplayTree() {
   const allWidgets = Array.from(widgets.values());
-  let result = 'CTRL-click a widget on\nthe left to edit it.\n\nRoom\n';
-  result += jeDisplayTreeAddWidgets(allWidgets, null, '  ');
+  $('#jeTree').innerHTML = '<ul class=jeTreeDisplay>' + jeDisplayTreeAddWidgets(allWidgets, null) + '</ul>';
+
+  $('#jeWidgetSearch').innerHTML = '<i class=extern>Search Room:  </i><input id="jeWidgetSearchBox" type="text"><div id="jeWidgetSearchResults"></div>';
```

Comment:
If you turn `<i class=extern>Search Room:  </i>` into `<label for="jeWidgetSearchBox" class=extern>Search Room:  </label>`, it will automatically be clickable and focus the search box.

Reply:
Fixed.